### PR TITLE
Replaced Mockito.mock by creating actual object

### LINKED ISSUE #3407: OpenLibertyUtilsTest : Replace `Mockito.mock` by creating actual object
### Component

JKube Kit

### Component

JKube Kit

### Component

JKube Kit

### Task description

Related to https://github.com/eclipse/jkube/issues/2316

OpenLibertyUtilsTest seems to be using mock here : 

https://github.com/eclipse-jkube/jkube/blob/dad4cc61149c435dbf91a5700a2c7f7cfe7d9a99/jkube-kit/jkube-kit-openliberty/src/test/java/org/eclipse/jkube/openliberty/OpenLibertyUtilsTest.java#L39

This can easily be refactored to use real object using lombok builders:

```java
      project = JavaProject.builder().build();
```
Similarly these `when(` statements can also be replaced with builder calls like this:

https://github.com/eclipse-jkube/jkube/blob/dad4cc61149c435dbf91a5700a2c7f7cfe7d9a99/jkube-kit/jkube-kit-openliberty/src/test/java/org/eclipse/jkube/openliberty/OpenLibertyUtilsTest.java#L40

```java
    javaProject = JavaProject.builder()
      .baseDirectory(temporaryFolder)
      .build();
```
Usage of `org.mockito.Mockito.mock`  and `org.mockito.Mockito.when` is removed from OpenLibertyUtilsTest
### Expected Behavior
Usages of `Mockito.mock` and `Mockito.when` are removed from OpenLibertyUtilsTest

### Acceptance Criteria
- [ ] mock statements are replaced via real variables

### PR DESCRIPTION
**Pull request by ShivamChavan01**
## Description

Replaced Mockito.mock by creating actual object 

Fixes #3407

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3414 (_2024-09-29T05:50:33Z_)

:gear: [JKube E2E Tests (11090152627)](https://github.com/eclipse-jkube/ci/actions/runs/11090152627) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3414)](https://github.com/eclipse-jkube/ci/actions/runs/11090152627/job/30812125391)

File: jkube-kit/jkube-kit-openliberty/src/test/java/org/eclipse/jkube/openliberty/OpenLibertyUtilsTest.java [hunk 1]
```diff
@@ -36,8 +36,10 @@ class OpenLibertyUtilsTest {
 
   @BeforeEach
   void setup() {
-    javaProject = mock(JavaProject.class, RETURNS_DEEP_STUBS);
-    when(javaProject.getBaseDirectory()).thenReturn(temporaryFolder);
+    javaProject = JavaProject.builder().build();
```
Comment (rohanKanojia) [inline jkube-kit/jkube-kit-openliberty/src/test/java/org/eclipse/jkube/openliberty/OpenLibertyUtilsTest.java:39]:
```suggestion
```

File: jkube-kit/jkube-kit-openliberty/src/test/java/org/eclipse/jkube/openliberty/OpenLibertyUtilsTest.java [hunk 2]
```diff
@@ -36,8 +36,9 @@ class OpenLibertyUtilsTest {
 
   @BeforeEach
   void setup() {
-    javaProject = mock(JavaProject.class, RETURNS_DEEP_STUBS);
-    when(javaProject.getBaseDirectory()).thenReturn(temporaryFolder);
+    javaProject = JavaProject.builder()
```
Comment (rohanKanojia) [inline jkube-kit/jkube-kit-openliberty/src/test/java/org/eclipse/jkube/openliberty/OpenLibertyUtilsTest.java:39]: Could you please remove unused imports as well?

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**